---
title: "APHL AMD Academy"
subtitle: "Microreact Activity"
author:
  - name: "Alexandra Lorentz"
    email: "alexandra.lorentz@dgs.virginia.gov"
    affiliations: "Virginia State Public Health Laboratory"
  - name: "Jared Johnson"
    email: "jared.johnson@doh.wa.gov"
    affiliations: "Washington State Public Health Laboratory"
format: 
  html:
    embed-resources: true
theme: flatly
title-block-banner: true
editor: visual
---

# Background

This activity will focus on a surveillance cluster of *Acinetobacter baumannii* from 2018-2024. This Microreact file was generated using [BigBacter](https://github.com/DOH-JDJ0303/bigbacter-nf) - a genomic surveillance tool that performs core SNP analyses and phylogenetic analysis.

# Set up (5 minutes)

1.  **Download the initial Microreact file:**

    Right-click this [link](https://raw.githubusercontent.com/DOH-JDJ0303/binfx-hub/refs/heads/main/docs/presentations/2025/APHL_AMD_Days/tool_selection/amd_academy_tool_selection_ex1.html) and save the file to your computer. The file should be called `file_name.microreact`.

2.  **Navigate to the Microreact webpage:**

    <https://microreact.org/upload>

3.  **Import the Microreact file:**

    Import the file downloaded in step 1 into Microreact.This can be accomplished by dragging the file into the browser or by clicking the large "plus" button on the lower right-hand corner of the Microreact webpage.

::: callout-note
The Microreact file should include three panels: a phylogenetic tree, a SNP matrix, and a metadata table.
:::

# Exploring Microreact functionality (10 minutes)

Take some time to explore the ways you can change the visualizations in Microreact and the data included in this file. There are many ways to explore the data and change the visualizations. If you are unfamiliar with Microreact, we suggest starting with the ‚ÄúEye‚Äù Icon to change the labels, colors, and shapes. You can also select certain samples, zoom in on parts of the tree, or otherwise explore the functionality.

# Updating the metadata (5 minutes)

The initial Microreact file included a phylogenetic tree, SNP matrix, and some sample metadata; however, this sample metadata is incomplete. The next steps will walk you through importing a new metadata file that contains some additional sample information. This information will be used during a later activity.

1.  **Download the new metadata file:**

    Right-click this [link](https://raw.githubusercontent.com/DOH-JDJ0303/binfx-hub/refs/heads/main/docs/presentations/2025/APHL_AMD_Days/tool_selection/amd_academy_tool_selection_ex1.html) and save the file to your computer. The file should be called `file_name.csv`.

2.  **Import the new metadata file into Microreact:**

    a.  Click the ‚Äúpencil‚Äù icon in the top right corner and select ‚ÄúEdit Existing Panels‚Äù.

    b.  Select the `Metadata` panel to edit.

    c.  Click the `file icon üìÑ` in the `Data File` field to upload the file downloaded in step 1.

::: callout-note
You should now see three new columns: `YEAR`, `MONTH`, and `DAY`.
:::

# Exploring the data (15 minutes)

The phylogenetic tree in the initial Microreact file was colored by `partitions`. BigBacter creates these partitions by estimating the pairwise nucleotide differences between samples based on the branch lengths in the tree.

::: callout-note
Make sure the `Color Column` is set to `PARTITION`. This can be changed using the `Eye icon üëÅÔ∏è`in the upper right-hand corner of Microreact page.
:::

::: {.callout-tip icon="false"}
## Questions

**Q1. Each parition includes samples that are within \~100 nucleotides of each other.**

a.  How many partitions only include 1 sample?

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

b.  What does this mean for those samples?¬†

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q2. Let‚Äôs explore the data in a single partition. Filter the metadata table to view only samples in Partition 8.**

a.  What sequence types (`MLST_*` columns) are included in this particular partition?

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

b.  Using the ‚ÄúShow Values‚Äù toggle under Labels in the SNP matrix, what is the SNP range within Partition 8?

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q3. Remove the filter for Partition 8 to view the full dataset. Change the Coulor Column to ‚ÄúMLST_1‚Äù.¬†Think about the differences between grouping the samples by partition and by MLST. Which do you find to be the most useful for identifying potential clusters?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```
:::

# Add in metadata blocks (15 minutes)

**INSERT BACKGROUND / INFO ABOUT WHY THIS IS RELEVANT**

1.  **Add AMR genes to tree:**

    Add metadata blocks for each antibiotic resistance gene using the `Metadata blocks` drop down menu in the `Tree` panel.

2.  **Inspect the AMR genes:**

    View the legend to interpret the colors in the metadata blocks.

3.  **Change the metadata block colors (optional)**

    You can change the color of the metadata blocks by selecting the `Pencil icon ‚úèÔ∏è` and select `Edit Existing Panels`. Colors can be adjusted in the `Metadata` table. Select the `Reuse` option if you want all color schemes to match.

::: {.callout-tip icon="false"}
## Questions

**Q1. What resistance genes are common? Which are uncommon?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q2. Reviewing the resistant gene data, are there any partitions or clusters of particular concern?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```
:::

# **Create a timeline (15 minutes)**

INSERT BACKGROUND

::: callout-note
This section requires the `YEAR`, `MONTH`, and `DAY` columns added in the `Updating the metadata` section of this activity.
:::

1.  **Creating a new panel:**

    a.  Click on the `Pencil icon ‚úèÔ∏è` in the upper right-hand corner of the screen and select `Create new Timeline`.
    b.  Use your cursor to select location for your new Timeline panel.
    c.  Add the `YEAR`, `MONTH`, `DAY` columns to the timeline.

2.  **Run the timeline:**

    a.  Click the `Play button ‚ñ∂Ô∏è` to start the timeline. Samples will appear based on their collection dates.

::: {.callout-tip icon="false"}
## Questions

**Q1. Was running the timeline helpful to understand the samples over time? What additional insights were gained by viewing and running the timeline?¬†**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```
:::

# **Investigate the recent samples (15 minutes?)**

1.  Annotate the _new_ samples:

    There are two samples that have recently been sequenced and added to the analysis. Color the tree nodes by `STATUS` to visualize these two samples on the tree.
    
::: {.callout-tip icon="false"}
## Questions

**Q1. Are these two samples related?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q2. Looking at the timeline, were there samples collected within a similar time frame?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q3. Zoom in on each of the new samples. What are the SNP differences between the new isolates and the other isolates closely related to each of these?**
```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

**Q4. How would you interpret these and communicate with infection prevention teams or local epidemiologists?**
```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```
:::

# **Saving and exporting the data (10 minutes)**

1.  Download the tree and metadata file:

    insert description

2.  Import the tree file into Auspice:

    insert description

3.  Import the metadata file into Auspice:

    insert description
    
::: {.callout-tip icon="false"}
## Questions

**Q1. What are the pros and cons of each visualization tool?**

```{=html}
<textarea rows="5" cols="85" placeholder="...."></textarea>
```

:::
